app/src/main/java/com/google/android/samples/socialite/ui/home/timeline/Timeline.kt

```diff
@@ -238,15 +242,22 @@ fun MetadataOverlay(modifier: Modifier, mediaItem: TimelineMediaItem) {
     ) {
         if (mediaItem.type == TimelineMediaType.VIDEO) {
             val mediaMetadataRetriever = MediaMetadataRetriever()
-            mediaMetadataRetriever.setDataSource(
-                LocalContext.current,
-                Uri.parse(mediaItem.uri),
-            )
+            val context = LocalContext.current.applicationContext
 
-            val duration =
-                mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)
-                    ?.toLong()
-            duration?.let {
+            val duration: State<Long?> = produceState<Long?>(initialValue = null) {
+                withContext(Dispatchers.IO) {
```

Comment:
are we doing this on the IO thread to reduce lag time when loading the metadata from a remote url?

if so, can we add a quick comment here explaining why we're doing this on the IO thread